### PR TITLE
Inventory workflow MInor fix

### PR DESCRIPTION
**Pull request by syed-khadeerahmed**
## Description
Minor bug fix in Inventory workflow manager 

## Type of Change
- [ yes ] Bug fix
- [ no ] New feature
- [ no ] Breaking change
- [ no ] Documentation update

## Checklist
- [ yes ] My code follows the style guidelines of this project
- [ yes ] I have performed a self-review of my own code
- [ yes ] I have commented my code, particularly in hard-to-understand areas
- [ yes ] I have made corresponding changes to the documentation
- [ yes ] My changes generate no new warnings
- [ no ] I have added tests that prove my fix is effective or that my feature works
- [ yes ] New and existing unit tests pass locally with my changes
- [ no ] Any dependent changes have been merged and published in downstream modules
- [ yes ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ yes ] Tasks are idempotent (can be run multiple times without changing state)
- [ no ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [ yes ] Playbooks are modular and reusable
- [ no] Handlers are used for actions that need to run on change

## Documentation
- [ yes ] All options and parameters are documented clearly.
- [ yes ] Examples are provided and tested.
- [ yes] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/inventory_workflow_manager.py [hunk 1]
```diff
@@ -1891,21 +1889,40 @@ def is_device_provisioned(self, device_id):
...
-        api_response = self.dnac._exec(
-            family="sda",
-            function='get_provisioned_devices',
-            params={
-                "networkDeviceId": device_id
-            }
-        )
-        provisioned_devices = api_response.get('response')
-        self.log("API response from 'get_provisioned_devices': {}".format(provisioned_devices), "DEBUG")
+                if prov_respone:
```
Comment (madhansansel) [inline plugins/modules/inventory_workflow_manager.py:1902]: spelling mistake

File: plugins/modules/inventory_workflow_manager.py [hunk 2]
```diff
@@ -1891,21 +1889,40 @@ def is_device_provisioned(self, device_id):
             if the device is currently provisioned and returns the appropriate
             status.
         """
+        if self.get_ccc_version_as_integer() <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
+            try:
+                prov_respone = self.dnac._exec(
```
Comment (madhansansel) [inline plugins/modules/inventory_workflow_manager.py:1894]: prov_response

File: plugins/modules/inventory_workflow_manager.py [hunk 3]
```diff
@@ -1891,21 +1889,40 @@ def is_device_provisioned(self, device_id):
...
+        else:
+            try:
+                api_response = self.dnac._exec(
+                    family="sda",
+                    function='get_provisioned_devices',
+                    params={
+                        "networkDeviceId": device_id
+                    }
+                )
+                provisioned_devices = api_response.get('response')
```
Comment (madhansansel) [inline plugins/modules/inventory_workflow_manager.py:1917]: is_provisioned

File: plugins/modules/inventory_workflow_manager.py [hunk 4]
```diff
@@ -1891,21 +1889,40 @@ def is_device_provisioned(self, device_id):
             if the device is currently provisioned and returns the appropriate
             status.
         """
+        if self.get_ccc_version_as_integer() <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
+            try:
+                prov_respone = self.dnac._exec(
+                    family="sda",
+                    function='get_provisioned_wired_device',
+                    op_modifies=True,
```
[the remark this answers is not in the document]
Comment (syed-khadeerahmed) [reply plugins/modules/inventory_workflow_manager.py:1897]: op_modifies = true only

File: plugins/modules/inventory_workflow_manager.py (hunk 3 above)
Comment (syed-khadeerahmed) [reply plugins/modules/inventory_workflow_manager.py:1917]: updated the code

File: plugins/modules/inventory_workflow_manager.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (madhansansel) [reply plugins/modules/inventory_workflow_manager.py:1897]: To get_provisioned_wired_device, why do we need to set this to True?

**Review by madhansansel — APPROVED** [review]

**Pull request merged**